File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/Async.java
package com.github.fernthedev.lightchat.core.api;

import java.lang.annotation.*;

/**
 * Used to show that the method is ASYNC
 */
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.METHOD})
@APIUsage
@Documented
public @interface Async {}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/Async.kt
package com.github.fernthedev.lightchat.core.api

/**
 * Used to show that the method is ASYNC
 */
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@APIUsage
@MustBeDocumented
annotation class Async 